src/native/corehost/test/mockhostfxr/test_c_api.c

```diff
@@ -0,0 +1,8 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// hostfxr.h is a public API. When included in .c files, it may fail to compile
```

Comment:
sounded like there might be one or two other headers that could benefit from similar protection?

Reply:
Absolutely! The three documented headers: https://learn.microsoft.com/en-us/dotnet/core/tutorials/netcore-hosting
```
#include <nethost.h>
#include <coreclr_delegates.h>
#include <hostfxr.h>
```

I'll fix it in main.